esx_service/vsan_policy.py

```diff
@@ -43,6 +44,87 @@ def create(name, content):
     return create_policy_file(filename, content)
 
 
+def update(name, content):
+    """
+    Update the content of an existing VSAN policy in the VSAN datastore.
+    Update the policy content in each VSAN object currently using the policy. If
+    a VSAN policy of the given name does not exist return an error string.
+    Return None on success.
+    """
+    path = policy_path(name)
+    if not path:
+        return ERROR_NO_VSAN_DATASTORE
+
+    err = update_policy_file_content(path, content)
+    if err:
+        return err
+
+    return update_vsan_objects_with_policy(name, content)
+
+
+def update_policy_file_content(path, content):
+    """
+    Update the VSAN policy file content. Return an error msg or None on success.
+    """
+    try:
+        with open(path) as f:
+            existing_content = f.read()
+    except OSError as e:
+        if not os.path.isfile(path):
+            return 'Error: Policy does not exist'
+        else:
+            return 'Error opening existing policy file: {0}'.format(e)
+
+    if existing_content.strip() == content.strip():
+            return 'Error: New policy is identical to old policy. Ignoring.'
+
+    # Create a temporary file so we don't corrupt an existing policy file
+    tmpfile = '{0}.tmp'.format(path)
```

Comment:
nit: will it be created in the same folder ? Maybe it's bettet to do mktemp() or whatever it's called in python, to avoid leftrovers if something fails ?